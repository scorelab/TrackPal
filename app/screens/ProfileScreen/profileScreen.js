import React, { Component } from "react";
import {
  View,
  Text,
  Image,
  ScrollView,
  Dimensions,
  ActivityIndicator
} from "react-native";
import HeaderBar from "../../components/HeaderBar/headerBar.js";
import styles from "./style.js";
import { Card, Button, Rating } from "react-native-elements";
import { f, database } from "../../../config/config.js";

let deviceHeight = Dimensions.get("window").height;

export default class ProfileScreen extends Component {
  state = {
    name: null,
    picUrl: null,
    bd: null,
    userID: null,
    loading: false,
    rating: 0
  };
  componentDidMount() {
    var user = f.auth().currentUser;
    
    if (user !== null) {
      this.setState({
        name: user.displayName,
        picUrl: user.photoURL,
        bd: user.email,
        loading: true,
        userID: user.uid
      });

      // get user ratings
      var that = this;
      database
        .ref("/users")
        .child(user.uid)
        .once("value", function(data) {
          that.setState({ rating: data.val().ratings });
        });
    }
  }

  signOut() {
    var that = this;
    f.auth()
      .signOut()
      .then(function() {
        console.log("sign out");
        that.props.navigation.navigate("Login");
      })
      .catch(function(error) {
        console.log("error");
      });
  }

  render() {
    const { name, picUrl, bd, userID } = this.state;

    if (this.state.loading === false) {
      return (
        <View style={styles.container}>
          <HeaderBar title={"Profile"} />
          <ScrollView style={{ width: "100%" }}>
            <View style={styles.container}>
              <ActivityIndicator size="large" color="#0000ff" />
            </View>
          </ScrollView>
        </View>
      );
    } else {
      return (
        <View style={styles.container}>
          <HeaderBar title={"Profile"} />
          <ScrollView style={{ width: "100%" }}>
            <View style={styles.container}>
              <View style={styles.coverImageArea}>
                <Image
                  style={styles.coverImage}
                  source={require("../../images/cover_photo.jpeg")}
                />
              </View>

              {picUrl !== null ? (
                <Image style={styles.profileImage} source={{ uri: picUrl }} />
              ) : (
                <Image
                  style={styles.profileImage}
                  source={require("../../images/user_image_1.jpg")}
                />
              )}

              <Card
                containerStyle={{
                  width: "90%",
                  height: deviceHeight * 0.58
                }}
              >
                <View style={styles.contentArea}>
                  <Text style={styles.nameFont}>{name}</Text>
                  <Text style={styles.cityFont}>{bd}</Text>
                </View>

                <View style={styles.ratingStyles}>
                  <Rating
                    imageSize={this.props.size}
                    readonly
                    startingValue={this.state.rating}
                  />
                </View>

                <View>
                  <Text>
                    Lorem Ipsum is simply dummy text of the printing and
                    typesetting industry. Lorem Ipsum has been the industry's
                    standard dummy text ever since the 1500s, when an unknown
                    printer took a galley of type and scrambled it to make a
                    type specimen book. It has survived not only five centuries,
                    but also the leap into electronic typesetting, remaining
                    essentially unchanged. It was popularised in the 1960s with
                    the release of Letraset sheets containing Lorem Ipsum
                    passages, and more recently with desktop publishing software
                    like Aldus PageMaker including versions of Lorem Ipsum.
                  </Text>
                </View>
                <Button
                  title="Sign Out"
                  containerStyle={{
                    paddingTop: deviceHeight * 0.1
                  }}
                  onPress={() => this.signOut()}
                />
              </Card>
            </View>
          </ScrollView>
        </View>
      );
    }
  }
}
