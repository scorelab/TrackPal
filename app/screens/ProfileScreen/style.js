import { StyleSheet, Dimensions } from "react-native";
let deviceWidth = Dimensions.get("window").width;
let deviceHeight = Dimensions.get("window").height;

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "flex-start",
    alignItems: "center",
    backgroundColor: "#e8e8e8",
    width: "100%"
  },

  coverImageArea: {
    // backgroundColor:'red',
    alignSelf: "center",
    height: "5%",
    width: "auto",
    position: "absolute",
    top: 10
  },

  coverImage: {
    height: 150
  },

  profileImage: {
    marginTop: deviceHeight * 0.3 - 100,
    width: 80,
    height: 80,
    borderRadius: 80
  },

  contentArea: {
    marginTop: 1,
    alignItems: "center",
  },
  nameFont: {
    fontSize: 18,
    lineHeight: 18,
    fontWeight: "bold"
  },
  cityFont: {
    fontSize: 16,
    lineHeight: 16
    // fontWeight:'bold'
  },
  input: {
    marginTop: deviceHeight * 0.01,
    marginLeft: deviceWidth * 0.01,
    height: 45,
    width: deviceWidth * 0.98,
    borderRadius: 5,
    backgroundColor: "rgba(149, 176, 183, 1)",
    marginBottom: 15,
    color: "#FFF",
    paddingHorizontal: 10
  },
  saveButton: {
    marginTop: 100,
    marginBottom: 100,
    height: 45
  },
  signoutButton: {
    paddingTop: 1000
  },
  ratingStyles: {
      margin: 15
  }
});
