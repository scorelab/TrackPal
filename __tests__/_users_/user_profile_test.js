import React from 'react';
import ProfileScreen from '../../app/screens/ProfileScreen/profileScreen';

import renderer from 'react-test-renderer';

jest.useFakeTimers();

test('renders correctly', () => {
  const tree = renderer.create(<ProfileScreen/>).toJSON();
  expect(tree).toMatchSnapshot();
});