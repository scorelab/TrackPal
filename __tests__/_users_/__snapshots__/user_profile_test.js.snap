// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e8e8e8",
      "flex": 1,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00aced",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 375,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Profile
      </Text>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#e8e8e8",
            "flex": 1,
            "justifyContent": "flex-start",
            "width": "100%",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#0000ff"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
